libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.cpp

```diff
@@ -153,6 +176,30 @@ void AP_BattMonitor_UAVCAN::handle_battery_info_aux(const BattInfoAuxCb &cb)
     _has_battery_info_aux = true;
 }
 
+void AP_BattMonitor_UAVCAN::handle_mppt_stream(const MpptStreamCb &cb)
+{
+    const bool use_input_value = (uint32_t(_params._options.get()) & uint32_t(AP_BattMonitor_Params::Options::MPPT_Use_Input_Value)) != 0;
+    const float voltage = use_input_value ? cb.msg->input_voltage : cb.msg->output_voltage;
+    const float current = use_input_value ? cb.msg->input_current : cb.msg->output_current;
+
+    // use an invalid soc so we use the library calculated one
+    const uint8_t soc = 101;
```

Comment:
Another not for this PR...
Using `soc = 101`  for an invalid soc is pre-existing, but I'm not a fan of it. Technically, of course the battery can never get above 100 soc by definition both in current UAVCAN standard and of course reality. But if the  system is misconfigured (wrong voltages, capacity, etc.) it could end up above 101 SOC.

I just looked some code for a  UAVCAN battery monitor  and it just sends out 100 if it computes SOC above 100. Instead IMO a device should send a "hey this is isn't right state using  127". Just thought as we look to future DroneCan stuff.

Reply:
I'm also not a fan of setting SOC=101 as the way to fallback to the calculated SOC.  At least there is a comment to clarify why it is done.

Reply:
It's unused anyway. Where we fetch it it gets ignored if we're an MPPT. I could set to 0? 127?

Reply:
Consider it for a future improvement and just merge now?

Reply:
For future.... certainly. Might as well keep 101 since then you have to change it other places too...

Reply:
Ok.. merging!